pkg/rpcclient/waiter/waiter.go

```diff
@@ -90,6 +90,12 @@ type PollingBased struct {
 	config  PollConfig
 }
 
+// Config is a unified configuration for [Waiter] implementations that allows to
+// customize awaiting behaviour.
+type Config struct {
+	PollConfig
```

Comment:
@roman-khimov, do you think we need to remove `PollConfig` structure?

Reply:
Can be kept this way, maybe event-driven will get some configuration too.